### ISSUE TITLE
`mvn verify` fails

### ISSUE THREAD

**Issue by svenefftinge** [vavr-io/vavr#2334]
In Gitpod the `mvn verify` fails with the following output:

```
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.4.4:script (default) on project vavr: wrap: java.lang.Exception: A NoSuchMethodError exception was thrown: scala.Predef$.refArrayOps([Ljava/lang/Object;)[Ljava/lang/Object; -> [Help 1]
```
I found [multiple references to similar issues](https://github.com/vavr-io/vavr/search?q=%22A+NoSuchMethodError+exception+was+thrown%3A+scala%22&type=Issues) which very closed as resolved. Here's my version info for mvn and java:
```
gitpod /workspace/vavr $ mvn -version
Picked up JAVA_TOOL_OPTIONS: -Xmx3063m
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z)
Maven home: /usr/share/maven
Java version: 1.8.0_191, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-1015-gcp", arch: "amd64", family: "unix"
```
Reproduce: [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod-staging.com#snapshot/64bd3117-c0c1-4053-a258-934d53a05c15)

**Comment by danieldietrich** [comment]
Hi Sven,

I hope you are doing well!

I think this issue has to do with a combination of

* Java version
* Maven version
* Maven Scala Maven Plugin version

We give a [hint](https://github.com/vavr-io/vavr/blob/master/CONTRIBUTING.md#building) in our CONTRIBUTION.md (but it is not obvious).

@jbrains gave a great [overview](https://github.com/vavr-io/vavr/issues/2321#issuecomment-435586146) in the issue #2321. Interestingly, he had no successful build with Java 8 at all. On my machine (macOS Mojava ) the build runs with all JDKs (8, 9, 10, 11) and Maven 3.5.4.

How could this be solved in [Gitpod](https://www.gitpod.io)?

**Comment by danieldietrich** [comment]
Yes, I will update the readme. What puzzles me is that it works with Java 8 in other environments. I have the impression, that the issue still isn't understood fully.

However, starting with Vavr 1.0 (which is in the make), we switch from Maven to Gradle. Then the issue will disappear...
